libeos-updater-util/avahi-service-file.h

```diff
@@ -44,4 +46,160 @@ gboolean eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
+ *
+ * - summary timestamp,
+ *   - key: "st",
+ *   - type: "t" (uint64)
+ *   - contents: a unix utc timestamp of the summary, ideally telling when
+ *               the original summary was created, otherwise it could also
+ *               be the modification time of the summary file on host
+ */
+
+/**
+ * EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y "force-version"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y "bloom-hash-id"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_BLOOM_K_Y:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_K_Y "bloom-k"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_BLOOM_SIZE_U:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_SIZE_U "bloom-size"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q "repository-index"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_PORT_Q:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_PORT_Q "port"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y "txt-records-size-level"
+/**
+ * EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_CUSTOM_SIZE_T:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_CUSTOM_SIZE_T "txt-records-custom-size"
+
+/* values for the EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y option */
+
+/**
+ * EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID ((guchar)1)
+
+/* values for the EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y option */
+
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_CUSTOM:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_CUSTOM ((guchar)0)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_SUPPORT_CRAPPY_HARDWARE:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_SUPPORT_CRAPPY_HARDWARE ((guchar)1)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_DNS_MESSAGE:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_DNS_MESSAGE ((guchar)2)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_ETHERNET_PACKET:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_ETHERNET_PACKET ((guchar)3)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_MULTICAST_DNS_PACKET:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_SINGLE_MULTICAST_DNS_PACKET ((guchar)4)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_16_BIT_LIMIT:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_FIT_16_BIT_LIMIT ((guchar)5)
+/**
+ * EOS_OSTREE_AVAHI_SIZE_LEVEL_ABSOLUTELY_LAX:
+ *
+ * TODO
+ */
+#define EOS_OSTREE_AVAHI_SIZE_LEVEL_ABSOLUTELY_LAX ((guchar)6)
+
+gboolean eos_ostree_avahi_service_file_check_options (GVariant  *options,
+                                                      GError   **error);
+gboolean eos_ostree_avahi_service_file_generate (const gchar   *avahi_service_directory,
+                                                 OstreeRepo    *repo,
```

Comment:
What do you think about replacing the `OstreeRepo *repo` parameter with a string list describing refs and a GDateTime describing repo's summary timestamp? I remember I tried to avoid using ostree API in previous version of DNS-SD records generation API. This would be also tons easier to test. Not sure how to pass them - should them be a part `GVariant *options` (say, "advertised-refs" being an `as` variant and "summary-timestamp" being a `t` variant or rather as explicit parameters to the function. I'm leaning to the latter, as refs and timestamp are not optional and have no sane default values we could use when the user does not specify "advertised-refs" or "summary-timestamp" options.